android/src/main/java/com/dieam/reactnativepushnotification/modules/RNPushNotificationJsDelivery.java

```diff
@@ -58,20 +58,29 @@ void notifyNotificationAction(Bundle bundle) {
     }
 
     String convertJSON(Bundle bundle) {
+        try {
+            JSONObject json = convertJSONObject(bundle);
+            return json.toString();
+        } catch (JSONException e) {
+            return null;
+        }
+    }
+    
+    // a Bundle is not a map, so we have to convert it explicitly
+    JSONObject convertJSONObject(Bundle bundle) throws JSONException {
         JSONObject json = new JSONObject();
         Set<String> keys = bundle.keySet();
         for (String key : keys) {
-            try {
-                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
-                    json.put(key, JSONObject.wrap(bundle.get(key)));
-                } else {
-                    json.put(key, bundle.get(key));
-                }
-            } catch (JSONException e) {
-                return null;
+            Object value = bundle.get(key);
+            if (value instanceof Bundle) {
```

Comment:
Are you sure that in our case notification bundle has nested 'Bundle's and not just string (which looks like bundle)?